torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3090,6 +3090,11 @@ Tensor log1p_backward(const Tensor& grad, const Tensor& self) {
   return grad / (self + 1).conj();
 }
 
+Tensor sinc_backward(const Tensor& grad, const Tensor& self) {
+  auto out = grad * ((M_PI * self * (M_PI * self).cos() - (M_PI * self).sin()) / (M_PI * self * self)).conj();
+  return at::where(self == 0.0, at::zeros({}, grad.options()), out);
```

Comment:
I mainly copied this from other functions in this file. But I'm not sure why `at::zeros({}, ...)` works and it doesn't have to be `at::zeros_like(...)`

Reply:
Yes that is perfect.
The zeros({}) call will create a scalar Tensor with a single 0 value. And at::where will automatically broadcast that to the size it needs.
This is better than the zeros_like because you don't actually need to allocate the memory for that full broadcasted Tensor. And the where kernel is faster because it doesn't need to read all these 0s from memory, just one.

Reply:
Thanks for the explanation.